ui/src/shared/components/RefreshingView.tsx

```diff
@@ -142,6 +132,10 @@ class RefreshingView extends PureComponent<Props, State> {
   private get variableAssignments(): VariableAssignment[] {
     const {timeRange, variableAssignments} = this.props
 
+    if (!timeRange) {
```

Comment:
Here you are using whether or not timeRange is defined here as a proxy for whether `v.timeRangeStart` exists in the query- but it will not make sense to the next programmer who looks at this code why timeRange and variableAssigments are related. I'm not sure you need to alter this code at all, but if you feel like you do - you should re-derive whether `v.timeRangeStart` exists in the query.

Reply:
I hear ya. I agree. It had originally errored out since the prop was initially set to `null`